action.yml

```diff
@@ -1,28 +1,23 @@
-# See: https://docs.github.com/en/actions/creating-actions
-name: TODO:Action Name
+name: RubyGems Setup GPR
 author: FreeAgent
-description: TODO:Description
-#  inputs:
-#    foo:
-#      description: Foo level
-#      required: true
-#      default: 23
-#  outputs:
-#    bar:
-#      description: Bar output
+description: Configure ruby gem to access for GPR (Github Package Registry)
+inputs:
+  token:
+    description: "Token to access the GitHub package registry. Normally using secrets.github_token is enough."
+    required: true
 
 runs:
-  # Pick one of the 3 possible types of action:
+  using: "composite"
+  steps:
+    - name: Setup GPR
+      shell: bash
+      env:
+        GEM_HOST_API_KEY: "Bearer ${{ inputs.token }}"
+      run: |
+        mkdir -p $HOME/.gem
+        touch $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+        printf -- "---\n:github: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
 
-  #using: 'composite'
-  #steps:
-  #  - name: Step One
-  #    shell: bash
-  #    run: |
-  #      echo TODO: Step One
-
-  #using: 'node12'
-  #main: main.js
-
-  #using: 'docker'
-  #image: 'Dockerfile'
+        echo "GEM_HOST=https://rubygems.pkg.github.com/${{ github.repository_owner }}" >> $GITHUB_ENV
+        echo "GEM_HOST_API_KEY=$GEM_HOST_API_KEY" >> $GITHUB_ENV
```

Comment:
I'm surprised that there isn't already an action for this, but I can't find one. I'm hoping that someone like the RubyGems organisation or GitHub will put out some, but it's no bad thing to just write our own. 

Closest I found, some of which do the job, but not from people I'd trust 👍 :
https://github.com/marketplace?query=gem+package

Since there aren't many others, let's get verified and have people use ours from the marketplace 🏅

Reply:
That was the idea.